### PR TITLE
Rewrite Parent event details without ReactiveSwift

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-14519
affects: Parent
release note: Updated the design of event details

### CODE REVIEW

**Comment by vanwagonet** [comment]
This also renamed CalendarEventItem -> CalendarEvent, since the conflicting CanvasCore version is removed.

Test Plan:
twilson observer1, student1 > Pages > Front Page > Event 1 (on student1 calendar) link at bottom
The event details should be presented much like they are for students.

Not that this removes the reminder capability for events, but considering I couldn't get to any from the Parent calendar, & needed to make a link in the home page to see one, I don't know if that's important.

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/hhpv7uk) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fhhpv7uk)
teacher: [Link](https://rebrand.ly/bmq5a93) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fbmq5a93)
parent: [Link](https://rebrand.ly/y8zoloi) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fy8zoloi)

**Comment by inst-danger** [comment]
Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.33% | 93.32% | 0.01%

[MBL-14519](https://instructure.atlassian.net/browse/MBL-14519)

#### Affected Apps: Parent

#### Release Note: 
Updated the design of event details
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by armstrongnate** [comment]
Edit: when I tap an event from the planner list in **parent** not _student_.

**Comment by vanwagonet** [comment]
I'd like to update the design a little & put reminders back in before merging.
Sounds like the parent planner isn't using the embedInNav option...

File: Parent/Parent/Extensions/UserNotificationReminder.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        let content = UNMutableNotificationContent()
+        content.title = NSLocalizedString("Event reminder", comment: "")
+        content.body = String.localizedStringWithFormat(
+            NSLocalizedString("%@ will begin at %@", comment: ""),
+            event.title,
+            event.startAt!.dateTimeString
+        )
+        content.userInfo = [
+            RemindableStudentIDKey: studentID,
+            RemindableActionURLKey: event.htmlURL!.absoluteString
```
Comment (inst-danger) [inline Parent/Parent/Extensions/UserNotificationReminder.swift:36]:
```suggestion
            RemindableActionURLKey: event.htmlURL!.absoluteString,
```

File: Parent/Parent/Extensions/UserNotificationReminder.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        let content = UNMutableNotificationContent()
+        content.title = NSLocalizedString("Assignment reminder", comment: "")
+        content.body = assignment.dueAt.map { String.localizedStringWithFormat(
+            NSLocalizedString("%@ is due at %@", comment: ""),
+            assignment.name,
+            $0.dateTimeString
+        ) } ?? assignment.name
+        content.userInfo = [
+            RemindableStudentIDKey: studentID,
+            RemindableActionURLKey: assignment.htmlURL!.absoluteString
```
Comment (inst-danger) [inline Parent/Parent/Extensions/UserNotificationReminder.swift:51]:
```suggestion
            RemindableActionURLKey: assignment.htmlURL!.absoluteString,
```

**Pull request merged**